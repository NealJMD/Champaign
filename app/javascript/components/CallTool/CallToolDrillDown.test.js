import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CallToolDrillDown from './CallToolDrillDown';
import { targets } from './call_tool_helpers.test';

const onUpdate = jest.fn();
const baseProps = {
  targets,
  onUpdate,
  targetByAttributes: ['country', 'state', 'name'],
  filters: {},
};

describe('CallToolDrillDown Snapshots', () => {
  it('renders filters when there are targetByAttributes', () => {
    const wrapper = shallow(<CallToolDrillDown {...baseProps} />);
    expect(toJSON(wrapper)).toMatchSnapshot('CallToolDrillDownWithFilters');
  });

  it('pre-selects the first property if it is passed in the filter', () => {
    const props = { ...baseProps, filters: { country: 'United States' } };
    const wrapper = shallow(<CallToolDrillDown {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot('CallToolDrillDownWithFilters');
  });

  it('pre-selects the first two properties if they passed in the filter', () => {
    const props = {
      ...baseProps,
      filters: { country: 'United States', state: 'California' },
    };
    const wrapper = shallow(<CallToolDrillDown {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot('CallToolDrillDownWithFilters');
  });

  it('pre-selects nothing if the passed filter is not first', () => {
    const props = { ...baseProps, filters: { state: 'California' } };
    const wrapper = shallow(<CallToolDrillDown {...props} />);
    expect(toJSON(wrapper)).toMatchSnapshot('CallToolDrillDownWithFilters');
  });
});
