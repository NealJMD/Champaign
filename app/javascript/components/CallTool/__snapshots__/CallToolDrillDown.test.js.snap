// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallToolDrillDownWithFilters 1`] = `
<div
  className="CallToolDrillDown"
>
  <div
    style={
      Object {
        "marginBottom": "10px",
      }
    }
  >
    <SweetSelect
      clearable={false}
      label="Country"
      name="filter-targets-by-attribute-country"
      onChange={[Function]}
      options={
        Array [
          Object {
            "id": "country-United States",
            "label": "United States",
            "value": "United States",
          },
          Object {
            "id": "country-Canada",
            "label": "Canada",
            "value": "Canada",
          },
        ]
      }
    />
    <div
      className="clearfix"
    />
  </div>
</div>
`;

exports[`CallToolDrillDownWithFilters 2`] = `
<div
  className="CallToolDrillDown"
>
  <div
    style={
      Object {
        "marginBottom": "10px",
      }
    }
  >
    <SweetSelect
      clearable={false}
      label="Country"
      name="filter-targets-by-attribute-country"
      onChange={[Function]}
      options={
        Array [
          Object {
            "id": "country-United States",
            "label": "United States",
            "value": "United States",
          },
          Object {
            "id": "country-Canada",
            "label": "Canada",
            "value": "Canada",
          },
        ]
      }
      value="United States"
    />
    <div
      className="clearfix"
    />
  </div>
  <div
    style={
      Object {
        "marginBottom": "10px",
      }
    }
  >
    <SweetSelect
      clearable={true}
      label="State"
      name="filter-targets-by-attribute-state"
      onChange={[Function]}
      options={Array []}
    />
    <div
      className="clearfix"
    />
  </div>
</div>
`;

exports[`CallToolDrillDownWithFilters 3`] = `
<div
  className="CallToolDrillDown"
>
  <div
    style={
      Object {
        "marginBottom": "10px",
      }
    }
  >
    <SweetSelect
      clearable={false}
      label="Country"
      name="filter-targets-by-attribute-country"
      onChange={[Function]}
      options={
        Array [
          Object {
            "id": "country-United States",
            "label": "United States",
            "value": "United States",
          },
          Object {
            "id": "country-Canada",
            "label": "Canada",
            "value": "Canada",
          },
        ]
      }
    />
    <div
      className="clearfix"
    />
  </div>
  <div
    style={
      Object {
        "marginBottom": "10px",
      }
    }
  >
    <SweetSelect
      clearable={true}
      label="State"
      name="filter-targets-by-attribute-state"
      onChange={[Function]}
      options={Array []}
    />
    <div
      className="clearfix"
    />
  </div>
  <div
    style={
      Object {
        "marginBottom": "10px",
      }
    }
  >
    <SweetSelect
      clearable={true}
      label="Name"
      name="filter-targets-by-attribute-name"
      onChange={[Function]}
      options={Array []}
    />
    <div
      className="clearfix"
    />
  </div>
</div>
`;

exports[`CallToolDrillDownWithFilters 4`] = `
<div
  className="CallToolDrillDown"
>
  <div
    style={
      Object {
        "marginBottom": "10px",
      }
    }
  >
    <SweetSelect
      clearable={false}
      label="Country"
      name="filter-targets-by-attribute-country"
      onChange={[Function]}
      options={
        Array [
          Object {
            "id": "country-United States",
            "label": "United States",
            "value": "United States",
          },
          Object {
            "id": "country-Canada",
            "label": "Canada",
            "value": "Canada",
          },
        ]
      }
    />
    <div
      className="clearfix"
    />
  </div>
</div>
`;
